.github/workflows/github-actions-essential-ci.yml

```diff
@@ -45,7 +49,9 @@ jobs:
     steps:
       - uses: actions/checkout@v4
         with:
-          ref: ${{ github.event.pull_request.head.sha }}
+          ref: ${{ github.event.pull_request.head.sha || github.ref }}
+      - name: compute metadata
+        run: echo GITHUB_ACTIONS_BRANCH=${{ github.event.pull_request.number || github.ref_name}} >> "$GITHUB_ENV"
```

Comment:
(just out of curiosity) what happens to `$GITHUB_ENV` (where is this used?)

Reply:
This is actually a built-in GitHub Actions feature: [Ref](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable)

Reply:
ah, TIL - thanks!